app/services/visit_timeline.rb

```diff
@@ -34,6 +34,10 @@ def events
 
 private
 
+  def visit_state_changes
+    @visit.visit_state_changes.sort { |a, b| a.created_at <=> b.created_at }
```

Comment:
Takes advantage of eager loaded data if it is in memory, and given that it will be a small number of objects to sort there is little gain to doing in sql